Source/FullScreenPokemon.ts

```diff
@@ -2977,6 +2977,34 @@ module FullScreenPokemon {
             return true;
         }
 
+        /**
+         * Collision callback for a Character and a WaterEdge. If possible, the Character
+         * is animated to move onto land.
+         *
+         * @param thing   A Character walking to other.
+         * @param other   A Ledge walked to by thing.
+         */
+        collideWaterEdge(thing: ICharacter, other: IThing): boolean {
+            if (!thing.surfing) {
+                return false;
+            }
+
+            if (other.title === "WaterEdgeTop" && thing.direction !== 0) {
```

Comment:
A better way to do this would be to define an allowed direction on each of the WaterEdge classes in objects.js, and check for it here. That way you avoid a large ugly set of if statements (or a large ugly switch statement).